src/core/thread/mle_router.cpp

```diff
@@ -1350,23 +1348,18 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
                 }
             }
         }
-        else
+        else if (router->mState != Neighbor::kStateValid)
         {
-            router = &mRouters[routerId];
-
-            if (router->mState != Neighbor::kStateValid)
-            {
-                memcpy(&router->mMacAddr, &macAddr, sizeof(router->mMacAddr));
-                router->mLinkInfo.Clear();
-                router->mLinkInfo.AddRss(mMac.GetNoiseFloor(), threadMessageInfo->mRss);
-                router->mLinkFailures = 0;
-                router->mState = Neighbor::kStateLinkRequest;
-                SendLinkRequest(router);
-                ExitNow(error = kThreadError_NoRoute);
-            }
+            memcpy(&router->mMacAddr, &macAddr, sizeof(router->mMacAddr));
+            router->mLinkInfo.Clear();
+            router->mLinkInfo.AddRss(mMac.GetNoiseFloor(), threadMessageInfo->mRss);
+            router->mLinkFailures = 0;
+            router->mState = Neighbor::kStateLinkRequest;
+            SendLinkRequest(router);
+            ExitNow(error = kThreadError_NoRoute);
         }
 
-        router->mLastHeard = Timer::GetNow();
+        mParent.mLastHeard = Timer::GetNow();
```

Comment:
should this line put into `if (memcmp(&mParent.mMacAddr, &macAddr, sizeof(mParent.mMacAddr)) == 0) `block?